modules/tinymce/src/plugins/noneditable/main/ts/api/Options.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Type } from '@ephox/katamari';
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorOptions } from 'tinymce/core/api/OptionTypes';
+
+const option: {
+  <K extends keyof EditorOptions>(name: K): (editor: Editor) => EditorOptions[K] | undefined;
+  <T>(name: string): (editor: Editor) => T | undefined;
+} = (name: string) => (editor: Editor) =>
+  editor.options.get(name);
+
+const register = (editor: Editor): void => {
+  const registerOption = editor.options.register;
+
+  registerOption('noneditable_noneditable_class', {
+    processor: 'string',
+    default: 'mceNonEditable'
+  });
+
+  registerOption('noneditable_editable_class', {
+    processor: 'string',
+    default: 'mceEditable'
+  });
+
+  registerOption('noneditable_regexp', {
+    processor: (value) => {
+      if (Type.isArray(value)) {
+        return { value, valid: true };
+      } else if (value?.constructor === RegExp) {
```

Comment:
Good point... I was pretty dead when doing these yesterday so I missed that. Really this probably needs to be something we add to Katamari so that we can use the cross frame checks we have there.

Reply:
I'll leave this as is for now seeing as it's the same as it was previously and log a follow up to fix this up and look at adding a `regexp` built-in processor.

Reply:
Have fixed this in the upcoming 7th PR for these changes: https://github.com/tinymce/tinymce/pull/7404